### PR TITLE
Disable middle mouse paste

### PR DESCRIPTION
**Pull request by olmokramer**
I keep accidentally pasting stuff by clicking the middle mouse button, so I wanted an option to disable it. This adds a new setting to the `editor` schema that toggles middle mouse paste.

### CODE REVIEW

**Comment by Zireael07** [comment]
+1. I keep accidentally doing it too.

**Comment by olmokramer** [comment]
@Zireael07 You can put this in your `init.coffee` until this is merged:
```coffee
atom.workspace.observeTextEditors (editor) ->
  editorComponent = atom.views.getView(editor).component
  {scrollViewNode} = editorComponent
  scrollViewNode.removeEventListener 'mousedown', editorComponent.onMouseDown
  scrollViewNode.addEventListener 'mousedown', (event) ->
    return if event.button is 1
    editorComponent.onMouseDown event
```

**Comment by steelbrain** [comment]
:+1:

**Comment by lee-dohm** [comment]
Would you mind adding some specs for this, @olmokramer?

**Comment by olmokramer** [comment]
Of course :). I wanted to know the general opinion on this before putting lots of effort in.

**Comment by lee-dohm** [comment]
I think it would be fine, personally.

**Comment by thedaniel** [comment]
Does this setting belong on all platforms, or just Linux?

**Comment by mehcode** [comment]
I'd probably rename it to `selectionCliboard` or `selectionPaste` as that's what X calls it.

**Comment by olmokramer** [comment]
@mehcode and it sounds a lot better :)

**Comment by thedaniel** [comment]
I'm sorry, I should have been more clear. [Middle mouse paste only *exists* on linux](https://github.com/atom/atom/blob/master/src/text-editor-component.coffee#L560-L563) so the setting doesn't make sense on other platforms.

**Comment by paulodiovani** [comment]
Middle mouse button is a GNU/Linux feature, but it doesn't exists on OSX and Linux.
I think it could be, by default, available or not, depending on OS.

**Comment by mnquintana** [comment]
> Middle mouse button is a GNU/Linux feature, but it doesn't exists on OSX and Linux.
I think it could be, by default, available or not, depending on OS.

@paulodiovani That's already the case: https://github.com/atom/atom/pull/6712/files#diff-4d704f98b832abd91d35923176b3c4d1R369

**Comment by paulodiovani** [comment]
If so, there's no reason to have a disable option.
You [cannot really disable middle mouse button on unix](http://unix.stackexchange.com/a/24346/27284).

**Comment by olmokramer** [comment]
Hi, I'm sorry it took so long, but I've been really busy with other stuff. I just pushed a new commit that seems to work. It doesn't actually disable the clipboard, just the pasting with the middle mouse button.

I'm trying to write some specs for it, but I don't know how to actually run them... I searched for it, but couldn't find out how. Can anyone point me in the right direction?

**Comment by olmokramer** [comment]
Forgot to mention, I found the "Run Atom specs" menu entry under `View -> Develop`, but that crashes instantly...

**Comment by mnquintana** [comment]
@olmokramer You can run core specs with `apm test`

**Comment by olmokramer** [comment]
> I'm sorry, I should have been more clear. Middle mouse paste only exists on linux so the setting doesn't make sense on other platforms.

The setting *is* available only on Linux systems.

> If so, there's no reason to have a disable option.
You cannot really disable middle mouse button on unix.

I'm not trying to disable the middle mouse button, I'm changing how Atom behaves when it is pressed.

**Comment by olmokramer** [comment]
@mnquintana Thanks!

File: src/text-editor-component.coffee [hunk 1]
```diff
@@ -227,7 +227,8 @@ class TextEditorComponent
     window.addEventListener 'resize', @requestHeightAndWidthMeasurement
 
     @listenForIMEEvents()
-    @trackSelectionClipboard() if process.platform is 'linux'
+    if process.platform is 'linux'
```
Comment (50Wliu) [inline src/text-editor-component.coffee]: Is there any specific reason why this was changed?

Comment (olmokramer) [inline src/text-editor-component.coffee]: I checked for the `editor.selectionClipboard` setting here, but decided that it wouldn't be necessary. The line was getting really long with both conditionals so I put it like this, and I forgot to revert the change when I removed it again.

Want me to change it back?

Comment (50Wliu) [inline src/text-editor-component.coffee]: Makes the diff smaller, I suppose :wink:.

Comment (olmokramer) [inline src/text-editor-component.coffee]: Ok, done :)

**Comment by olmokramer** [comment]
Ok I think this is ready for review. Thanks for the help and feedback so far! :)

**Comment by davidak** [comment]
Why is this still not merged?

**Comment by wI2L** [comment]
UP

**Comment by robclancy** [comment]
Please tell me this is done in some way already and this issue is just old and not forgotten. I keep coming across super basic issues in Atom that shouldn't be a thing in any editor.

**Comment by atomdmac** [comment]
+1

**Comment by davidak** [comment]
@50Wliu

**Comment by 50Wliu** [comment]
@davidak Why am I being @-mentioned exactly?

**Comment by davidak** [comment]
@50Wliu because i saw that you are very active with merging PRs so you may can answer my question from some months ago.

This bug prevents me from using Atom like SublimeText with multiselections. Here is a PR to potentially fix it so it would be quiet sad if it's laying around for months.

**Comment by 50Wliu** [comment]
I may merge lots of PRs, but most of those are either on language repos or otherwise contain changes that I'm familiar with and can test extensively.  Neither of those two apply here 🙂.

As for it not being merged, we should be getting to it soon.  Sorry for the delay.

**Comment by rushrtb2112** [comment]
What's the latest on this?

**Comment by jronallo** [comment]
The snippet for init.coffee in https://github.com/atom/atom/pull/6712#issuecomment-100472802 no longer seems to work in 1.19.0. I get a `Cannot read property 'removeEventListener' of undefined`. Is there a current solution for disabling middle mouse paste? Extra points for a solution that will disable middle mouse paste in code but not Markdown.

**Comment by balderclaassen** [comment]
For anyone struggling with this, the earlier mentioned StackExchange [link](https://unix.stackexchange.com/questions/24330/how-can-i-turn-off-middle-mouse-button-paste-functionality-in-all-programs/301772#301772) shows that in several Linux distributions you can now easily turn off the middle mouse paste behaviour with gnome-tweak-tools. Of course, that doesn't work if you want to use middle mouse paste outside of Atom.
I personally have a middle mouse button that simply isn't that sensitive so I didn't even know middle mouse paste existed.

**Comment by envygeeks** [comment]
On Ubuntu

```
xinput set-button-map 9 1 0 3
xinput set-button-map 10 1 0 3
```

**Comment by WyRe** [comment]
Is still planned to merge this branch? I guess so, but it has been a long time until last work ...

**Comment by robclancy** [comment]
Haha is this still not done? Thank god I moved on. Does atom at least have a refresh option in the sidebar now?

**Review by ghost — APPROVED** [review]

**Comment by sadick254** [comment]
Closing in favour of https://github.com/atom/atom/pull/21864

**Comment by robclancy** [comment]
> robclancy on Feb 14, 2019
> Haha is this still not done? Thank god I moved on.
hahahaha
